gammapy/detect/cwt.py

```diff
@@ -73,25 +73,29 @@ class CWT(object):
     background model to which was added the flux of the extragalactic background,
     and the exposure for the considered energy range were used.
 
+    Read more in ???.
 
     TODO: instead of storing all the arrays as data members,
     we could have a ``.data`` member which is a dict of images?
 
     Parameters
     ----------
-    min_scale : float
-        first scale used
-    nscales : int
-        number of scales considered
-    scale_step : float
-        base scaling factor
+    n_scale : int
+        Number of scales considered.
+    min_scale : double
+        First scale used.
+    scale_step : double
+        Base scaling factor.
     max_iter : int, optional (default 10)
         The maximum number of iterations os the CWT algorithm.
     tol : double, optional (default 1e-5)
         Tolerance for stopping criterion.
-
-    TODO: give references
-    Read more in ???.
+    nsigma (significance max???) : double, optional (default 3.0)
```

Comment:
@OlgaVorokh - `nsigma` and `nsigmap` are measures of statistical significance, "multiples of sigma" or "multiples of standard deviation":
https://en.wikipedia.org/wiki/Statistical_significance#Stringent_significance_thresholds_in_specific_fields

In other parts of Gammapy that quantity is called "significance", e.g. here you can see how it relates to a p-value:
https://gammapy.readthedocs.io/en/latest/api/gammapy.stats.significance_to_probability_normal.html

Maybe rename to `nsigma` to `significance_threshold` and `nsigmap` to `significance_threshold_clean`?
The second threshold is only used for the cleaning of isolated pixel islands that are slightly above `significance_threshold`.
That's a non-standard thing to do and should be made optional.
Maybe have `significance_threshold_clean=None` by default and unless the user gives a float number for that, don't do the pixel island cleaning.

Does this help?